src/plugins/gs-flatpak.c

```diff
@@ -653,24 +687,66 @@ gs_flatpak_add_updates (GsFlatpak *self, GsAppList *list,
 			continue;
 		}
 
-		/* we have an update to show */
-		g_debug ("%s has a downloaded update %s->%s",
-			 flatpak_ref_get_name (FLATPAK_REF (xref)),
-			 commit, latest_commit);
 		app = gs_flatpak_create_installed (self, xref, &error_local);
 		if (app == NULL) {
-			g_warning ("failed to add flatpak: %s", error_local->message);
+			g_warning ("failed to add downloaded update for "
+				   "flatpak %s: %s",
+				   flatpak_ref_get_name (FLATPAK_REF (xref)),
+				   error_local->message);
 			continue;
 		}
-		if (gs_app_get_state (app) == AS_APP_STATE_INSTALLED)
-			gs_app_set_state (app, AS_APP_STATE_UNKNOWN);
+
+		/* we have an update to show */
+		g_debug ("%s has a downloaded update %s->%s",
+			 gs_app_get_unique_id (app),
+			 commit, latest_commit);
+
+		gs_app_set_state (app, AS_APP_STATE_UNKNOWN);
 		gs_app_set_state (app, AS_APP_STATE_UPDATABLE_LIVE);
 		gs_app_list_add (list, app);
 	}
 
 	return TRUE;
 }
 
+gboolean
+gs_flatpak_add_updates (GsFlatpak *self, GsAppList *list,
+			GCancellable *cancellable,
+			GError **error)
+{
+	guint i;
+	g_autoptr(GPtrArray) xrefs = NULL;
+
+	/* manually drop the cache */
+	if (0&&!flatpak_installation_drop_caches (self->installation,
```

Comment:
Good question! I'd rather not change that as part of these changes and fix it separately after checking with Richard.